### PR TITLE
GHA/windows: enable ECH feature on WolfSSL

### PR DESCRIPTION
**Pull request by talregev**
Enable ECH feature on WolfSSL.

### CODE REVIEW

**Comment by talregev** [comment]
It not compile: 

```
D:\a\curl\curl\lib\vtls\wolfssl.c(1242,22): error C2039: 'hinfo': is not a member of 'Curl_dns_entry' [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
      D:\a\curl\curl\lib\hostip.h(95,8):
      see declaration of 'Curl_dns_entry'
  
D:\a\curl\curl\lib\vtls\wolfssl.c(1243,21): error C2037: left of 'echconfiglist' specifies undefined struct/union 'Curl_https_rrinfo' [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vtls\wolfssl.c(1244,[32](https://github.com/curl/curl/actions/runs/12035110446/job/33553325322?pr=15647#step:9:32)): error C2037: left of 'echconfiglist' specifies undefined struct/union 'Curl_https_rrinfo' [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vtls\wolfssl.c(1245,25): error C2037: left of 'echconfiglist_len' specifies undefined struct/union 'Curl_https_rrinfo' [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
```

**Comment by vszakats** [comment]
Did you also enable `-DUSE_HTTPSRR=ON`?

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -622,7 +622,7 @@ jobs:
             config: >-
               -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DCURL_USE_LIBSSH2=ON
               -DCURL_USE_SCHANNEL=OFF -DCURL_USE_WOLFSSL=ON -DUSE_NGTCP2=ON
-              -DCURL_USE_GSASL=ON
+              -DCURL_USE_GSASL=ON -DUSE_HTTPSRR=ON -DUSE_ECH=ON
```
Comment (vszakats) [inline .github/workflows/windows.yml:625]:
```suggestion
              -DCURL_USE_GSASL=ON
              -DUSE_HTTPSRR=ON -DUSE_ECH=ON
```
Sync formatting with BoringSSL.

**Comment by talregev** [comment]
> Did you also enable `-DUSE_HTTPSRR=ON`?

Now I enable and it working. 

It make sense that if user enable ech, it automatically also enable httpsrr?

File: .github/workflows/windows.yml (hunk 1 above)
Comment (talregev) [reply .github/workflows/windows.yml]: Done.

**Comment by vszakats** [comment]
> > Did you also enable `-DUSE_HTTPSRR=ON`?
> 
> Now I enable and it working.
> 
> It make sense that if user enable ech, it automatically also enable httpsrr?

It most likely would, esp. since ECH doesn't work without RR.

PR → 15648

**Pull request closed without merging**